### ISSUE TITLE
Identical builds works w/ docker build, fails with docker-compose (1.4.2)

### ISSUE THREAD

**Issue by tarikihm** [docker/compose#2109]
Hi, 

I have the following docker-compose.yml:
```
web:
  build: .
  volumes:
   - app:/web/app
```

which fails with 
```
Running "compass:dist" (compass) task
directory dist/css
    write dist/css/auth.css (1.791s)
    error app/client/sass/not-supported.scss (Line 2: No files were found in the load path matching "notsupp/*.png". Your current load paths are: /web/./app/client/assets/img)
Sass::SyntaxError on line ["2"] of /web/./app/client/sass/not-supported.scss: No files were found in the load path matching "notsupp/*.png". Your current load paths are: /web/./app/client/assets/img
Run with --trace to see the full backtrace
Warning: ↑ Use --force to continue.

Aborted due to warnings.
Service 'web' failed to build: The command '/bin/sh -c grunt build' returned a non-zero code: 6
```

The normal build, i.e.: `docker build -t ihr/web .` doesn't fail.

**Comment by dnephin** [comment]
`volumes` are not used during the build.

Could you provide the `Dockerfile` you're using, as well as the command you ran?

**Comment by tarikihm** [comment]
@dnephin I know, I am not using it during the build, it can be removed from the docker-compose.yml as it is optional, the same build issue happens.

The part of the Dockerfile that fails is:

`
# Build app assets
RUN grunt build
`
Command I ran is `docker-compose up`

It works when running `docker build -t ihr/web .` (same Dockerfile)

I think if I provide you the full Dockerfile, you may then want to look at the source code, so not sure it would help? But shouldn't be an issue. Either way why would the normal docker build work and the same docker build with the same options and Dockerfile fails when docker-compose runs it?

**Comment by dnephin** [comment]
Are you using `.dockerignore`? We recently fixed some issues in the `docker-py` implementation of `.dockerignore`. They're on master, but aren't in a release yet.

If you're using `.dockerignore` try out master to see if it resolves the issue.

**Comment by tarikihm** [comment]
Yes, I am using `.dockerignore`. I can wait for the next release, but glad to try on master. Would I just need the latest version of docker/compose (I assumer it's what using `docker-py`) and how to I update it on a OS X Docker Toolbox install?

**Comment by thaJeztah** [comment]
@tarikihm is this still an issue? Are you still able to reproduce this with the current release, or can this issue be closed?

**Comment by tarikihm** [comment]
yes, issue is fixed as of Docker Toolbox 1.9.0, thank you!

**Comment by lucaspottersky** [comment]
I've just stumbled upon this issue when creating a new Digital Ocean Droplet:

```
Docker version 1.11.2, build b9f10c9
docker-compose version 1.7.1, build 0a9ab35
```

Dockerfile:
```
FROM python:2.7
RUN pip install scrapy scrapy-splash scrapyd scrapyd-client
```

Works if I just `docker build .`, but fails when I `docker-compose build`:

```
  Running setup.py bdist_wheel for lxml: finished with status 'error'
```
